bundles/framework/timeseries/WMSAnimator.js

```diff
@@ -0,0 +1,180 @@
+    }, {
+        __name: 'WMSAnimator',
+        getName: function () {
+            return this.__name;
+        },
+        _clazz: 'Oskari.mapframework.bundle.timeseries.WMSAnimator',
+        __eventHandlers: {
+            'ProgressEvent': function (event) {
+                if (event.getStatus() && this._layer.getId() === event.getId()) {
+                    this._isLoading = false;
+                    this._resolveWait();
+                }
+            }
+        },
+        onEvent: function (event) {
+            var handler = this.__eventHandlers[event.getName()];
+            if (!handler) {
+                return;
+            }
+            return handler.apply(this, [event]);
+        },
+        /**
+         * @method getTimes
+         * Returns all available time instants that can be shown
+         * @return {String[]} list of available timeseries times
+         */
+        getTimes: function () {
+            var times = this._layer.getAttributes().times;
+            if (!Array.isArray(times)) {
+                var interval = moment.duration(times.interval);
+                var end = moment(times.end);
+                var t = moment(times.start);
+                times = [t.toISOString()];
+                while (t.add(interval) < end) {
+                    times.push(t.toISOString());
+                }
+                times.push(end.toISOString());
+            }
+            return times;
+        },
+        init: function () { },
+        /**
+         * @method getCurrentTime
+         * Returns current selected time instant
+         * @return {String} current time as ISO-string
+         */
+        getCurrentTime: function () {
+            return this._currentTime;
+        },
+        /**
+         * @method getSubsetRange
+         * Returns current selected time range subset
+         * @return {String[]} range Array with 2 elements, start & end, ISO-string
+         */
+        getSubsetRange: function () {
+            return this._subsetRange.slice();
+        },
+        /**
+         * @method getSubsetRange
+         * Set current selected time range subset
+         * @param {String[]} range Array with 2 elements, start & end, ISO-string
+         */
+        setSubsetRange: function (range) {
+            this._subsetRange = range;
+        },
+        /**
+         * @method requestNewTime
+         * Requests change in current selected time
+         * @param {String} newTime change current time to this value(ISO string)
+         * @param {String} nextTime time value at next animation frame(ISO string). Can be null if not animating
+         * @param {function} doneCallback callback that will be called after new time has been loaded
+         */
+        requestNewTime: function (newTime, nextTime, doneCallback) {
+            var me = this;
+            this._currentTime = newTime;
+            var requestBuilder = this._sandbox.getRequestBuilder('MapModulePlugin.MapLayerUpdateRequest');
```

Comment:
Use Oskari.requestBuilder() instead as sandbox.getRequestBuilder() has been deprecated. If you want to know if someone is actually handling the request you can check with sandbox.hasHandler('MapModulePlugin.MapLayerUpdateRequest').